grafast/grafserv/src/servers/deno.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  convertHandlerResultToResult,
+  GrafservBase,
+  normalizeRequest,
+} from "grafserv";
+
+import type {
+  GrafservBodyJSON,
+  GrafservConfig,
+  RequestDigest,
+  Result,
+} from "grafserv";
+
+declare global {
+  namespace Grafast {
+    interface RequestContext {
+      ctx: Deno.ServeHandlerInfo;
+    }
+  }
+}
+
+function getDigest(req: Request, ctx: Deno.ServeHandlerInfo): RequestDigest {
+  return {
+    httpVersionMajor: 2,
+    httpVersionMinor: 0,
+    isSecure: new URL(req.url).protocol === "https",
+    method: req.method,
+    path: new URL(req.url).pathname,
+    headers: Object.fromEntries(req.headers.entries()),
+    getQueryParams() {
+      return Object.fromEntries(new URL(req.url).searchParams) as Record<
+        string,
+        string | string[]
+      >;
+    },
+    async getBody() {
+      return {
+        type: "json",
+        json: await req.json(),
+      } as GrafservBodyJSON;
+    },
+    requestContext: { ctx },
+  };
```

Comment:
We should avoid calling `new URL()` many times when we could call it just once. I've also added a couple TODOs but they don't need to be addressed in this PR.

```suggestion
  const url = new URL(req.url);
  return {
    // TODO: figure out the actual HTTP version
    httpVersionMajor: 2,
    httpVersionMinor: 0,
    isSecure: url.protocol === "https",
    method: req.method,
    path: url.pathname,
    headers: Object.fromEntries(req.headers.entries()),
    getQueryParams() {
      // TODO: on duplicate keys, this should convert to an array instead
      return Object.fromEntries(url.searchParams) as Record<
        string,
        string | string[]
      >;
    },
    async getBody() {
      return {
        type: "json",
        json: await req.json(),
      } as GrafservBodyJSON;
    },
    requestContext: {
      denov1: {
        ctx,
      },
    },
  };
```